platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpMetadata.java

```diff
@@ -161,9 +206,56 @@ private EntityDescriptor getEntityDescriptor() {
       boolean updated = entryDescriptions.compareAndSet(null, edMap);
       if (!updated) {
         LOGGER.debug("Safe but concurrent update to serviceProviders map; using processed value");
+      } else {
+        setCacheInformation(extractRootEntityFromMap(edMap));
       }
     }
 
+    return extractRootEntityFromMap(edMap);
+  }
+
+  private void setCacheInformation(EntityDescriptor entityDescriptor) {
```

Comment:
✏️  `entityDescriptor` should be annotated with `@Nullable`
❓ Is there any concerns for multi-threading to account for here?

Reply:
This class has always had multithreading problems (and should be rewritten eventually probably).

Reply:
On further consideration, there is a possible multi-threading issue. If two threads are attempting to set the IDP metadata, and they each have different metadata, the object could end up with the `created` value from one thread and the `validUntil` value from another thread. 
The metadata is only changed when an administrator changes it. In practice two almost simultaneous updates should always have the same metadata. However, I will add `synchronize` the method if @rzwiefel or @paouelle want it.

Reply:
❗️ If more than one thread will be using these values (and others in this class) then you must either synchronize access (read or write) to them or make them volatile or atomic. Failure to do so means one thread could be accessing a stale value This is true even if you are guaranteed that the value is modified before being read. If done by 2 different threads, and those are running on different processors then you are not properly flushing or re-reading the processor's cache.